### ISSUE TITLE
Annotation data for C2C12 dataset

### ISSUE THREAD

**Issue by yfukai** [talbenha/cell-tracker-gnn#1]
Hi, I truly appreciate uploading the code to GitHub with detailed instructions!
I was wondering if it is possible to ask to upload the ground-truth segmentation data used for the C2C12 benchmark, as it would make it much easier for other researchers to compare their methods with this GNN-based method. I understand the [original data](https://osf.io/ysaq2/) is published with the CC-By license, but it only provides limited segmentation annotations.

**Comment by talbenha** [comment]
Hello @yfukai,
The C2C12 benchmark provides the cell coordinate annotations according to this form: <frame_index>  <cell_id> <col_index> <row_index> <parent id>
Parent ID shows cell division information. Daughter cells have parent cell id. If the cell has not divided, the cell has -1.


As mentioned in the paper, I followed the training scheme introduced in the paper "MPM: Joint Representation of Motion and Position Map for Cell Tracking". 

I am adding the answer they gave me in an email when I asked them about their training:
> 
> We used 4 sequences.
>  1. Sequence 9 of Dataset01 (3rd March 2009). We used 100 images from 680 to 780. 
>  2.  Sequence 2 of Dataset01 (3rd March 2009). We used 100 images from 600 to 700.
>  3. Sequence 17 of Dataset 02 (18th March 2009). We used 100 images from 600 to 700.
>  4. Sequence 18 of Dataset 02 (18th March 2009). We used 100 images from 600 to 700.
> The annotation of sequence 9 is publicly available from this website https://www.nature.com/articles/sdata2018237.pdf .
> The other 3 sequences have been uploaded to github https://github.com/JunyaHayashida/MPM/tree/master/additional_annotation

> The annotation of sequence 17 is given from 600 frames to 700 frames. But, the start and end frames seems edited to 0 to 100.  Please add 600 for each frame_index.


From these annotations, you can get your training data, the rest are partially annotated and used for the test.
Hope it helps you.

**Comment by yfukai** [comment]
Hello @talbenha, thank you for guiding me to the dataset! That is helpful information to reproduce the benchmark.
So if I understand correctly, the selected frames from 
- sequence 17 and 18 for FGF2+BMP
- sequence 9 for BMP2 (originally provided in the dataset article)
- sequence 2 for control

are used for training, and 
- the rest of the frames of those datasets
- the other datasets

are used as the test dataset.

About the test set, I found the segmentation was generated as
> We annotated the test sequences of the C2C12 using the nnU-Net network proposed in [18].

I suppose you generated the segmentation for all the cells in the field. Was this segmented data manually validated? Could I ask if it is uploaded somewhere?

**Comment by talbenha** [comment]
Hi @yfukai,
Yes - the segmentations were generated for all the cells in the field. Where the nnU-Net was trained using the same data mentioned above. 
No - this segmented data was't manually validated.

It is really huge data so I cannot upload it right now.

**Comment by yfukai** [comment]
Thanks for your answer @talbenha! 
- Could I ask you to share the trained model to reproduce the result?
- If not, could I ask you to upload the training script?

> here the nnU-Net was trained using the same data mentioned above.

I was a bit confused since nnU-net is a segmentation model, as far as I understand. How did you get the segmentation mask from the centroid markers?

**Comment by talbenha** [comment]
Hi @yfukai 
From the centroid coordinate I had generated a circle with predefined radius and this artifact was used as GT - we didn't use pixel level GT to train the segmentation model since we don't have, we used roughly estimated markers to serve as cell detections. 

We trained the nnU-net using their documentation (highly recommended to go though) and we specifically used their recommendation to the cell tracking challenge 
- here you can find notebook that explains how they trained the model for cell microscopy data https://github.com/MIC-DKFZ/nnUNet/blob/nnunetv1/documentation/celltrackingchallenge/MIC-DKFZ.ipynb
and you can get easily trained model using their guideline.

Let me know if you have more questions.

**Comment by yfukai** [comment]
Thanks @talbenha for your reply! Unfortunately we gave up on reproducing the benchmark due to its difficulty (Still there're unknown parameters for the segmentation and we can't make it sure it's working in the same way as done in the publication.) 
I still believe making the segmentation notebook public is the easiest way to ensure the reproducibility. Anyway I was inspired by this tracking method. Thanks for making the repository for the code!

**Comment by talbenha** [comment]
You're welcome! I'm glad to hear that you found inspiration in the tracking method. If there's anything else I can do to assist you, please don't hesitate to ask. Closing this issue for now :)

**Comment by 19904465540** [comment]
Where should this dataset exist, can you talk about the process of running it manually with pycharm?

**Comment by talbenha** [comment]
@19904465540 
The dataset is located here: https://osf.io/ysaq2/
what do you mean by:
> can you talk about the process of running it manually with pycharm?
